### ISSUE TITLE
Gui crash when scan network(SMB) disks.

### ISSUE THREAD

**Issue by caoli5288** [0x90d/videoduplicatefinder#237]
**Environment**
 - OS: Windows

**Describe the bug**
Gui frozen and crash when scan network disks. It's happens on 3.0 only, not on 2.0.

**Comment by 0x90d** [comment]
I cannot reproduce this on my mapped drive. I need more infos.

**Comment by caoli5288** [comment]
@0x90d How can I got crash report?

**Comment by caoli5288** [comment]
@0x90d It's crash at enums files step. > 100k videos in my NAS.

**Comment by 0x90d** [comment]
Maybe there is a crash entry in the event viewer

**Comment by caoli5288** [comment]
@0x90d Any idea?

**Comment by 0x90d** [comment]
I already wrote: Maybe there is a crash entry in the event viewer

**Comment by caoli5288** [comment]
@0x90d So any fix in progress?

**Comment by Maltragor** [comment]
Could it be that you two are talking past each other right now?
In that case, I'll try to resolve it:

caoli5288, you asked: 
> How can I got crash report?

0x90d answered:
> Maybe there is a crash entry in the event viewer

The _event viewer_ is a windows program:
https://www.howtogeek.com/123646/htg-explains-what-the-windows-event-viewer-is-and-how-you-can-use-it/

Or to put it another way: 
VDF does not have its own special crash report mechanism. However, Windows itself logs program crashes to a certain extent together with the program's error output. And depending on the conditions that have led to a crash, VDF sometimes also issues further error messages. 
=> Please look in the _event viewer_ in the corresponding section and if there is a current error entry from VDF, post it here.


_At least that would be my guess for why the rest of the conversation turns out a bit weird ;-)_

**Comment by caoli5288** [comment]
@Maltragor THX for your help! I got those infos. @0x90d 

```xml
- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name=".NET Runtime" /> 
  <EventID Qualifiers="0">1026</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>0</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2021-11-03T18:52:28.7835373Z" /> 
  <EventRecordID>37045</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="4808" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>LAPTOP-RQ5T48SI</Computer> 
  <Security /> 
  </System>
- <EventData>
  <Data>Application: VDF.GUI.exe CoreCLR Version: 6.0.21.48005 .NET Version: 6.0.0-rc.2.21480.5 Description: The process was terminated due to an unhandled exception. Exception Info: System.ArgumentOutOfRangeException: Not a valid Win32 FileTime. (Parameter 'fileTime') at System.DateTime.FromFileTimeUtc(Int64 fileTime) at VDF.Core.FileEntry..ctor(String file) in D:\a\videoduplicatefinder\videoduplicatefinder\VDF.Core\FileEntry.cs:line 41 at VDF.Core.ScanEngine.<BuildFileList>b__46_0() in D:\a\videoduplicatefinder\videoduplicatefinder\VDF.Core\ScanEngine.cs:line 142 at System.Threading.Tasks.Task.InnerInvoke() at System.Threading.Tasks.Task.<>c.<.cctor>b__271_0(Object obj) at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) --- End of stack trace from previous location --- at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) --- End of stack trace from previous location --- at VDF.Core.ScanEngine.StartSearch() in D:\a\videoduplicatefinder\videoduplicatefinder\VDF.Core\ScanEngine.cs:line 91 at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__127_0(Object state) at Avalonia.Threading.AvaloniaSynchronizationContext.<>c__DisplayClass5_0.<Post>b__0() in /_/src/Avalonia.Base/Threading/AvaloniaSynchronizationContext.cs:line 33 at Avalonia.Threading.JobRunner.RunJobs(Nullable`1 priority) in /_/src/Avalonia.Base/Threading/JobRunner.cs:line 37 at Avalonia.Win32.Win32Platform.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam) in /_/src/Windows/Avalonia.Win32/Win32Platform.cs:line 210 at Avalonia.Win32.Interop.UnmanagedMethods.DispatchMessage(MSG& lpmsg) at Avalonia.Win32.Win32Platform.RunLoop(CancellationToken cancellationToken) in /_/src/Windows/Avalonia.Win32/Win32Platform.cs:line 157 at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken) in /_/src/Avalonia.Base/Threading/Dispatcher.cs:line 61 at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 116 at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 148 at VDF.GUI.Program.Main(String[] args) in D:\a\videoduplicatefinder\videoduplicatefinder\VDF.GUI\Program.cs:line 31</Data> 
  </EventData>
  </Event>
```

**Comment by Maltragor** [comment]
OK, the interesting section is:

> Description: 
> The process was terminated due to an unhandled exception. 
> 
> Exception Info: 
> System.ArgumentOutOfRangeException: Not a valid Win32 FileTime. 
> (Parameter 'fileTime') at System.DateTime.FromFileTimeUtc(Int64 fileTime) at VDF.Core.FileEntry..ctor(String file) in D:\a\videoduplicatefinder\videoduplicatefinder\VDF.Core\FileEntry.cs:line 41

This means that (at least) one of your video files has an invalid creation or modification date.
The attempt to convert this file date into another format fails, but VDF does not catch this explicitly.


@caoli5288 
Depending on how much of a hurry you're in, and how you want to handle erroneous timestamps independently of VDF:

You could use a suitable tool to check and 'correct' the timestamps of your files. Since I have never used such a program, here is just the suggestion to use '_Fix Invalid File Date Time_' as a search term.  (I got several hits, which you could have a closer look at).

**Comment by Maltragor** [comment]
Quick note on fix #237:
VDF will now not crash on files with incorrect timestamps, but the files will be ignored with respect to the duplicate search. (The respective file name is output in the logging tab of VDF). 
=> If one wants that such files are considered with the search, one must thus first correct the timestamps of these files.
